dlt/extract/source.py

```diff
@@ -135,24 +136,31 @@ class DltResource(Iterable[TDataItem], DltResourceSchema):
 
     Empty: ClassVar["DltResource"] = None
 
-    def __init__(self, pipe: Pipe, table_schema_template: TTableSchemaTemplate, selected: bool):
+    def __init__(self, pipe: Pipe, table_schema_template: TTableSchemaTemplate, selected: bool, incremental: Incremental[Any, Any] = None):
         # TODO: allow resource to take name independent from pipe name
         self.name = pipe.name
         self.selected = selected
         self._pipe = pipe
+        self.incremental = incremental
+        if self.incremental:
+            self.incremental.resource_name = self.name
+            # TODO: Incremental needs to go last, I think. What happens when other steps are added after this?
```

Comment:
it probably should, but I'd ignore this problem for now. we could construct pipes dynamically when they are passed to `PipeIterator.from_pipes` or add some `factory` steps in the pipe itsef (like the `ForkPipe` I use to pump data across dependend resources)